chia/rpc/wallet_rpc_api.py

```diff
@@ -842,14 +843,20 @@ async def create_offer_for_ids(self, request):
             )
         if success:
             return {
-                "offer": trade_record.offer.hex(),
+                "offer": Offer.from_bytes(trade_record.offer).compress().hex(),
                 "trade_record": trade_record.to_json_dict_convenience(),
             }
         raise ValueError(error)
 
     async def get_offer_summary(self, request):
         assert self.service.wallet_state_manager is not None
         offer_hex: str = request["offer"]
+        try:
+            offer_hex = bytes(Offer.from_compressed(hexstr_to_bytes(request["offer"]))).hex()
+        except CompressionVersionError:
+            raise "This offer file is incompatible with your version of software.  Upgrade your software and try again."
+        except Exception:
+            pass
```

Comment:
Do we really want to ignore all other exceptions?

Reply:
I changed it to be ValueErrors only in an upcoming patch.